### ISSUE TITLE
iOS app calling 'initWithToken' why is the result hardcoded and no errors are thrown?

### ISSUE THREAD

**Issue by Catpants** [hoxfon/react-native-twilio-programmable-voice#32]
Please include following information for better support
When calling 'initWithToken' with an invalid token it does not throw an error. Also, when TwilioVoice.connect fails to connect it does not throw an error either, how are we suppose to react to errors? I see the errors in the logs. I am not sure if I am missing something.

What version of React Native are you running?
16.0.0-alpha.6

What version of react-native-twilio-programmable-voice are you running?
2.8.0

What device are you using? (e.g iOS9 simulator, Android 6 device)?
iPhone 6 device

Is your app running in foreground, background or not running?
foreground

Is there any relevant message in the log?
```
native[1019:281452] providerDidBegin
native[1019:283459] StartCallAction transaction request successful
native[1019:283411] StartCallAction transaction request failed: The operation couldn’t be completed. (com.apple.CallKit.error.requesttransaction error 7.)
native[1019:281452] provider:performStartCallAction
native[1019:281452] provider:didActivateAudioSession
native[1019:283053] [ERROR TVONotificationClient] Registration failed: Error Domain=com.twilio.voice.error Code=20101 "Invalid access token" UserInfo={NSLocalizedDescription=Invalid access token}
native[1019:283053] [ERROR TwilioVoice] Inside register:deviceToken:completion:, failed to register for Twilio push notifications. Error:Invalid access token
native[1019:283053] An error occurred while registering: Invalid access token
```

If using iOS, which pod version are you using?
1.3.1

Step to reproduce
Call TwilioVoice.initWithToken with an invalid token.

Advanced:
Have you tried adding break point in native handlers and see the logs printed?
can share a project with issue?
Did you try to reinstall the pods completely?

The more context you provide around this issue the faster the community can help you.

**Comment by fabriziomoscon** [comment]
@Catpants I fixed this with npm `3.1.0`
Basically I made the iOS token initialization process the same as on Andorid.
To see if there are errors you must listen to `deviceReady` and `deviceNotReady` events.

Please let me know if that works

**Comment by Catpants** [comment]
'deviceReady' does not seem to be called at all on v3.1.1 even though the token is valid when passed into 'initWithToken'. This is with iOS. All other issues seem to be resolved. Thanks for the awesome updates.
